### ISSUE TITLE
CameraControl Smart Pivot

### ISSUE THREAD

**Issue by xeolabs** [xeokit/xeokit-sdk#533]
![102763527-1908a480-437a-11eb-807f-45f632fc0dbe](https://user-images.githubusercontent.com/83100/104718973-0b9dca80-572c-11eb-9c3f-5201aec6d458.jpeg)

**Comment by xeolabs** [comment]
/ping @Amoki & @Kurtil  - I coded this in https://github.com/xeokit/xeokit-sdk/tree/improve-camera-pivot

Could you give it a try and see what you think?

I made the radius ````100```` by default, but it's configurable:

````javascript
viewer.cameraControl.defaultPivotSphereRadius = 70.0;
````

You could try it using the ````examples/CameraControl_orbit_Duplex.html```` example.

**Comment by Amoki** [comment]
It works really really well with one small update!!!

I set dynamically the radius with 
```javascript
const pivotShereRadius = math.distVec3(this._scene.center, camera.eye);
```

So it works very with both a building and a city!

**Comment by xeolabs** [comment]
Very interesting tweak...I'm testing this out with a few different combinations of models. Could I get the JSON metadata for the Lyon models?

**Comment by Amoki** [comment]
We don't use xeokit JSON metadata, we use your own :/

**Comment by xeolabs** [comment]
I just did a test where I loaded both Lyon and Duplex into the same scene, which results in a huge distance from camera eye to scene center. Therefore, when click-pivoting empty space, the sphere then ends up being very large, and all the objects get whisked out of view. 

So, it works nicely for a single "site" (eg. Lyon) but not for a scene containing multiple sites that are far from each other.

![Peek 2021-01-15 15-20](https://user-images.githubusercontent.com/83100/104737980-3b0d0100-5745-11eb-9cbf-e714faa69f45.gif)

Perhaps an option to switch between dynamic and configured sphere radius modes?
